tendrl/monitoring_integration/alert/handlers/__init__.py

```diff
@@ -0,0 +1,107 @@
+import importlib
+import inspect
+import json
+import os
+import six
+
+from tendrl.commons.event import Event
+from tendrl.commons.message import ExceptionMessage
+from tendrl.commons.utils import log_utils as logger
+from tendrl.monitoring_integration.alert.exceptions import AlertNotFound
+from tendrl.monitoring_integration.alert.exceptions import Unauthorized
+from tendrl.monitoring_integration.alert import utils
+from tendrl.monitoring_integration.grafana.exceptions import ConnectionFailedException
+from tendrl.monitoring_integration.objects.alert_types import AlertTypes
+
+class NoHandlerException(Exception):
+    pass
+
+
+class HandlerMount(type):
+
+    def __init__(cls, name, bases, attrs):
+        if not hasattr(cls, 'handlers'):
+            cls.handlers = []
+        else:
+            cls.register_handler(cls)
+
+    def register_handler(cls, handler):
+        instance = handler()
+        cls.handlers.append(instance)
+
+
+@six.add_metaclass(HandlerMount)
+class AlertHandler(object):
+    handles = ''
+    representive_name = ''
+
+    def __init__(self):
+        self.alert = None
+
+    def handle(self, alert_json):
+        alert = self.format_alert(alert_json)
+        logger.log( 
+            "notice",
+            NS.publisher_id,
+            {
+                "message" : json.dumps(alert)
+            }
+        )
+
+
+class AlertHandlerManager(object):
+    def _load_handlers(self):
+        path = os.path.dirname(os.path.abspath(__file__))
+        pkg = 'tendrl.monitoring_integration.alert.handlers'
+        handlers = utils.list_modules_in_package_path(path, pkg)
+        for name, handler_fqdn in handlers:
+            mod = importlib.import_module(handler_fqdn)
+            clsmembers = inspect.getmembers(mod, inspect.isclass)
+            for name, cls in clsmembers:
+                if issubclass(cls, AlertHandler) and cls.handles:
+                    self.alert_handlers.append(cls.handles)
+
+    def __init__(self):
+        self.alert_handlers = []
+        self._load_handlers()
+        self._save_alert_types()
+
+    def _save_alert_types(self):
+        alert_types = {}
+        for handler in AlertHandler.handlers:
+            alert_types[handler.representive_name] = "true"
```

Comment:
Default value is "true" for representative_name ?

Reply:
i will change this as array